src/features/defi/providers/idle/components/IdleManager/Deposit/IdleDeposit.tsx

```diff
@@ -0,0 +1,135 @@
+import { Center, useToast } from '@chakra-ui/react'
+import { toAssetId } from '@shapeshiftoss/caip'
+import { DefiModalContent } from 'features/defi/components/DefiModal/DefiModalContent'
+import { DefiModalHeader } from 'features/defi/components/DefiModal/DefiModalHeader'
+import {
+  DefiAction,
+  DefiParams,
+  DefiQueryParams,
+  DefiStep,
+} from 'features/defi/contexts/DefiManagerProvider/DefiCommon'
+import { useIdle } from 'features/defi/contexts/IdleProvider/IdleProvider'
+import qs from 'qs'
+import { useCallback, useEffect, useMemo, useReducer } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { useSelector } from 'react-redux'
+import { CircularProgress } from 'components/CircularProgress/CircularProgress'
+import { DefiStepProps, Steps } from 'components/DeFi/components/Steps'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import { useBrowserRouter } from 'hooks/useBrowserRouter/useBrowserRouter'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import {
+  selectAssetById,
+  selectMarketDataById,
+  selectPortfolioLoading,
+} from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import { Approve } from './components/Approve'
+import { Confirm } from './components/Confirm'
+import { Deposit } from './components/Deposit'
+import { Status } from './components/Status'
+import { IdleDepositActionType } from './DepositCommon'
+import { DepositContext } from './DepositContext'
+import { initialState, reducer } from './DepositReducer'
+
+export const IdleDeposit = () => {
+  const { idleInvestor } = useIdle()
+  const [state, dispatch] = useReducer(reducer, initialState)
+  const translate = useTranslate()
+  const toast = useToast()
+  const { query, history, location } = useBrowserRouter<DefiQueryParams, DefiParams>()
+  const chainAdapterManager = getChainAdapterManager()
+  const { chainId, contractAddress: vaultAddress, assetReference } = query
+
+  const assetNamespace = 'erc20'
+  const assetId = toAssetId({ chainId, assetNamespace, assetReference })
+  const asset = useAppSelector(state => selectAssetById(state, assetId))
+  const marketData = useAppSelector(state => selectMarketDataById(state, assetId))
+
+  // user info
+  const chainAdapter = chainAdapterManager.get(chainId)
```

Comment:
```suggestion
  const chainAdapter = chainAdapterManager.get(chainId) as ChainAdapter<KnownChainIds.EthereumMainnet>
```

Reply:
Or alternatively 
```suggestion
  const chainAdapter = chainAdapterManager.get(KnownChainIds.EthereumMainnet)
```

As you've been doing in other components, this only supports mainnet for now so no need to be programmatic and lose type safety